libs/expo/betterangels/src/lib/screens/Client/Docs/UploadModal/FileUploadTab.tsx

```diff
@@ -0,0 +1,70 @@
+import { CheckIcon, PlusIcon } from '@monorepo/expo/shared/icons';
+import { Colors, Radiuses, Spacings } from '@monorepo/expo/shared/static';
+import { Button } from '@monorepo/expo/shared/ui-components';
+import { StyleSheet, View } from 'react-native';
+import { Docs, ITab } from './types';
+
+export interface IFileUploadTabProps {
+  docs: Docs;
+  setTab: (tabKey: ITab) => void;
+  tabKey: keyof Docs;
+  title: string;
+}
+
+const FileUploadTab = (props: IFileUploadTabProps) => {
+  const { docs, setTab, tabKey, title } = props;
+  const isMultiUpload = Array.isArray(docs[tabKey as keyof Docs]);
+  const disabled = !isMultiUpload && !!docs[tabKey as keyof Docs];
```

Comment:
Perhaps also define what `docs[tabKey as keyof Docs` is?
Thinking something like

```javascript
  const uploadValue = docs[tabKey as keyof Docs];

  const isMultiUpload = Array.isArray(uploadValue);
  const disabled = !isMultiUpload && !!uploadValue];
```

Reply:
my threshold for storing something in a var is: (1) used more than twice; OR (2) used more than once and far apart from eachother. I'm not committed to this approach, so I can be convinced otherwise

Reply:
sounds good. one other consideration I have is that if it's not immediately clear what something like `docs[tabKey as keyof Docs]` is, then it's helpful to define it with a const.

Reply:
🤔 yea that's a good one